dev-util/codeblocks/codeblocks-20.03-r6.ebuild

```diff
@@ -0,0 +1,87 @@
+# Copyright 1999-2023 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+WX_GTK_VER="3.0-gtk3"
+
+inherit autotools flag-o-matic wxwidgets xdg
+
+DESCRIPTION="The open source, cross platform, free C, C++ and Fortran IDE"
+HOMEPAGE="https://codeblocks.org/"
+LICENSE="GPL-3"
+SLOT="0"
+KEYWORDS="~amd64 ~ppc ~x86"
+SRC_URI="mirror://sourceforge/${PN}/${P}.tar.xz
+https://dev.gentoo.org/~leio/distfiles/${P}-fortran.tar.xz
+https://dev.gentoo.org/~leio/distfiles/${P}-fortran-update-v1.7.tar.xz
+https://dev.gentoo.org/~leio/distfiles/${P}-fortran-update-v1.8.tar.xz
+https://dev.gentoo.org/~leio/distfiles/${P}-codecompletion-symbolbrowser-update.tar.xz
+"
+
+# USE="fortran" enables FortranProject plugin (updated to v1.8 2021-05-29 [r230])
+# that is delivered with Code::Blocks 20.03 source code.
+# https://sourceforge.net/projects/fortranproject
+# https://cbfortran.sourceforge.io
+
+IUSE="contrib debug fortran"
+
+BDEPEND="virtual/pkgconfig"
+
+RDEPEND="app-arch/zip
+	>=dev-libs/tinyxml-2.6.2-r3
+	>=dev-util/astyle-3.1-r2:0/3.1
+	x11-libs/wxGTK:${WX_GTK_VER}[X]
+	contrib? (
+		app-admin/gamin
+		app-text/hunspell
+		dev-libs/boost:=
+	)"
```

Comment:
@juippis some packages are already required by other dependencies (see below).  `app-arch/bzip2` could be a standalone requirement or provided either via `dev-libs/boost[bzip2]`, or `app-arch/zip[bzip2]`.  About `bzip2`, there's also the package `app-alternatives/bzip2` to take into consideration.  What do you think?

- With `WX_GTK_VER="3.0-gtk3"` in the ebuild, `dev-libs/glib:2`, `x11-libs/gtk+:3`, and `sys-libs/zlib` are already required by `x11-libs/wxGTK:${WX_GTK_VER}[X]`;
- `media-libs/fontconfig` is already required by `x11-libs/gtk+:3`;
- `dev-libs/libgamin` is already required by `app-admin/gamin` (under USE flag `contrib` section);
- `app-arch/bzip2` could be provided by `dev-libs/boost[bzip2]` (under USE flag `contrib` section).

Reply:
We don't rely on indirect deps, always list what your package needs. Although I'll admit gtk is a bit special since usual consumers of gtk will support whatever gtk was built with.

Reply:
Thanks for the explanation.  I updated the ebuild.